core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -328,6 +330,7 @@ private[spark] class MemoryStore(blockManager: BlockManager, maxMemory: Long)
     val droppedBlocks = new ArrayBuffer[(BlockId, BlockStatus)]
 
     accountingLock.synchronized {
+      releasePendingUnrollMemoryForThisThread()
```

Comment:
@andrewor14  
If put it at the end of `accountingLock.synchronized`, as I described above, 

```
accountingLock.synchronized {
  1. ensureFreeSpace() //here will call the currentUnrollMemory, it will re-count the pendingUnrollmemmory to get freeMemory, then to judge whether free space or not.
   2. entries.put(blockId, entry)
}
```

If still need to put it at the end of `accountingLock.synchronized`, for get free space in ensureFreeSpace.

may change the current code `val actualFreeMemory = freeMemory - currentUnrollMemory` to `val actualFreeMemory = freeMemory - currentUnrollMemory + unrollPendingMmeory(threadId)`